docker-entrypoint.sh

```diff
@@ -267,6 +275,51 @@ docker_sql_escape_string_literal() {
 	echo "${escaped//\'/\\\'}"
 }
 
+validate_replica_user() {
+	if [ "$MARIADB_REPLICATION_MODE" = "master" ]; then
```

Comment:
last fix - validation of environment variables should be in the function `docker_verify_minimum_env`.

Reply:
Simplification, remove `MARIADB_REPLICATION_MODE`. I think its possible. Note if you disagree and why.

Can `-n $MARIADB_MASTER_HOST` imply replica mode and absence imply master? The replication user is only created in master mode.

docker_verify_minimum_env:
```
if [ -n "$MARIADB_REPLICATION_USER" ]; then
   if [ -z "$MARIADB_MASTER_HOST" ] && [ -z "$MARIADB_REPLICATION_PASSWORD" ]; then
      mysql_error "A replica requires MARIADB_REPLICATION_PASSWORD"
   elif [ -z "$MARIADB_REPLICATION_PASSWORD" ] && [ -z "$MARIADB_REPLICATION_PASSWORD_HASH" ]; then
      mysql_error "A replication master requires MARIADB_REPLICATION_PASSWORD or MARIADB_REPLICATION_PASSWORD_HASH for the MARIADB_REPLICATION_USER" 
   fi
fi
```

Reply:
Did the same with some diffrent logic, Please review, If everything is fine then I'll have rebase/sqush my commits.

Reply:
The logic that you mentioned above requires to update all the tests with `MARIADB_REPLICATION_USER` & `MARIADB_REPLICATION_PASSWORD`.

Reply:
And removing all `MARIADB_REPLICATION_MODE`, yes it will. From the tests and entrypoint and the tests around validity need to be rewritten, however will be more comprehensive due to simplicity.

Reply:
The above `docker_verify_minimum_env` hasn't been added.